tgui/packages/tgui/interfaces/common/sorting/SortableTable/type.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license ISC
+ */
+
+import { BooleanLike } from "common/react";
+import { InfernoNode } from "inferno";
+import { SortDirection } from "../type";
+
+/** Configuration for Sortable Table header row and callback functions */
+export interface SortableTableHeaderConfig {
+
+  /** Presentation of header cell for this field */
+  children: InfernoNode,
+
+  /** Is this field sortable? */
+  sortable: BooleanLike,
+
+  /** Is this field searchable */
+  searchable: BooleanLike,
+
+  /** Optional tooltip string. */
+  toolTipContent?: string,
+
+  /** Callback function to sort on type specific to data given */
+  compareFunc?: (a: unknown, b: unknown) => number,
+
+  /** Conversion from your data to string to allow searching */
+  toString?: (data: unknown) => string,
```

Comment:
Could make use of generics here.
```suggestion
  /** Callback function to sort on type specific to data given */
  compareFunc?: (a: TValue, b: TValue) => number,

  /** Conversion from your data to string to allow searching */
  toString?: (data: TValue) => string,
```
Then at the top of this thing:
```
export interface SortableTableHeaderConfig<TValue>
```
You might hit some interesting Typescript errors when doing this (either here or wherever this is consumed). Happy to help you work through them if needed.